### ISSUE TITLE
adding code example to "Automatic return"

### ISSUE THREAD

**Issue by jnvaughn** [swcarpentry/r-novice-inflammation#374]
Would be nice to see something like this examples associated with the side note:

So, an alternative way of writing the previous function would be:
fahrenheit_to_kelvin <- function(temp_F) {
  ((temp_F - 32) * (5 / 9)) + 273.15
}

**Comment by katrinleinweber** [comment]
Good point, but I wonder whether we can turn [that side note](http://swcarpentry.github.io/r-novice-inflammation/02-func-R/#automatic-returns) into something even more useful ;-) How about a challenge that leads learners to the solution you posted above. They would have to figure out then, that the intermediate `temp_K` assignment can be removed as well. What do you think?

Either way: do you want to start a [PR about this against `02-func-R.Rmd`](https://github.com/swcarpentry/r-novice-inflammation/edit/gh-pages/_episodes_rmd/02-func-R.Rmd)? Either with your initial suggestion, or the challenge idea?

**Comment by engal** [comment]
I'm curious whether it's actually worth the time to make this point in the episode. I've found that some people familiar with other languages before learning R can find R code using automatic returns somewhat unintuitive. I think part of this is because functions without return statements can look like functions that are supposed to be modifying objects in place (which rarely happens in R, data.tables being the main exception I've encountered, but does happen in other languages).

Since there is no functional difference between an automatic return and using a return statement , might it be simpler to leave out the note about automatic returns (or just avoid spending additional time)? I think the main counterargument would be that it could help learners when they encounter other peoples' code with automatic returns, but this seems like a marginal gain for people trying to initially get started with R.

**Comment by swulfing** [comment]
Would it be useful to also include a quick note about assigning a global variable within a function (i.e. explain what '<<-' does within a function vs. '<-' ?). In my experience, this issue doesn't come up too often in R coding but because of that, when my students try to assign a global variable in a function, they have no idea what that is or how to do it. Discussions about a global vs local variable is a lot more common in other languages so more experienced coders may have already have a sense for how to use it.

It's not something that I think we should spend a lot of time on, I just think it will help novices recognize the difference and give more advanced coders the operator in case they need it.
